types/google-protobuf/google/protobuf/any_pb.d.ts

```diff
@@ -5,12 +5,12 @@ import * as jspb from "../../index";
 
 export class Any extends jspb.Message {
   getTypeUrl(): string;
-  setTypeUrl(value: string): void;
+  setTypeUrl(value: string): Any;
```

Comment:
Shouldn’t every one of these return types be `this`, so that subclassing works as expected? Adding some tests would be nice since this is an important popular package.

Reply:
> Shouldn’t every one of these return types be this, so that subclassing works as expected

I'm was not familiar with any subclassing problems, but to my knowledge the protobuf javascript generator is not written with sub-classing in mind:

reference: https://developers.google.com/protocol-buffers/docs/reference/javascript-generated#message
>  the protocol buffer compiler generates a class called Foo. Foo inherits from jspb.Message.
You should not create your own Foo subclasses. Generated classes are not designed for subclassing and may lead to "fragile base class" problems.

protoc plugins for typescript type generators also only return the type (which is not to say it's the source of truth)

https://github.com/agreatfool/grpc_tools_node_protoc_ts#book_pbdts
https://github.com/agreatfool/grpc_tools_node_protoc_ts/pull/55

Reply:
If the implementation does `return this` in the method (which it surely does), the correct return type is `this`. But naming the class instead might discourage subclassing, and if that’s to be desired I’m fine with leaving it as is.

Reply:
> If the implementation does `return this` in the method (which it surely does), the correct return type is `this`. But naming the class instead might discourage subclassing, and if that’s to be desired I’m fine with leaving it as is.

if someone feels strongly about returning `this` instead of the class, one could open another PR. the good thing is that it would be a non-breaking change.

thank you for the review @andrewbranch !